lib/pacemaker/pcmk_ticket.c

```diff
@@ -123,6 +123,118 @@ pcmk_ticket_constraints(xmlNodePtr *xml, const char *ticket_id)
     return rc;
 }
 
+static int
+delete_single_ticket(xmlNode *child, void *userdata)
+{
+    int rc = pcmk_rc_ok;
+    cib_t *cib = (cib_t *) userdata;
+
+    rc = cib->cmds->remove(cib, PCMK_XE_STATUS, child, cib_sync_call);
+    rc = pcmk_legacy2rc(rc);
+
+    return rc;
+}
+
+int
+pcmk__ticket_delete(pcmk__output_t *out, cib_t *cib, pcmk_scheduler_t *scheduler,
+                    const char *ticket_id, bool force)
+{
+    int rc = pcmk_rc_ok;
+    xmlNode *state = NULL;
+
+    CRM_ASSERT(cib != NULL && scheduler != NULL);
```

Comment:
I prefer to return EINVAL for libpacemaker API invalid args, so external callers have a chance of handling it

Reply:
I think we're okay asserting on these arguments, however.  They are set up inside the public function, not passed from the user.  So the only way they'd be NULL here is if that setup code failed (which is an error on our parts) or if some other caller in internal code passed NULL.  Both seem like our failures, which an assertion seems appropriate for.

But I can change them (and the previous spots I'm doing this) instead if that's what we are going for.

Reply:
Ah right, I neglected that